src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ScalarHMMSegmenter.java

```diff
@@ -0,0 +1,98 @@
+    }
+
+    @Override
+    protected boolean hiddenStateValuesHaveConverged(final List<Double> oldHiddenStateValues) {
+        final double[] oldHiddenStateArray = Doubles.toArray(oldHiddenStateValues);
+        final double[] newHiddenStateArray = Doubles.toArray(hiddenStateValues);
+        return oldHiddenStateArray.length == newHiddenStateArray.length &&
+                GATKProtectedMathUtils.maxDifference(oldHiddenStateArray, newHiddenStateArray) < CONVERGENCE_THRESHOLD;
+    }
+
+    // filter out components that have low weight and are too close to another component -- these will
+    // die out eventually in EM, but very slowly, so we hasten their demise for quicker convergence
+    @Override
+    protected void pruneUnusedComponents() {
+        final int K = weights.length;
+
+        final Set<Integer> componentsToPrune = new TreeSet<>();
+        for (final int state : IntStream.range(0, K).filter(n -> n != NEUTRAL_VALUE_INDEX).toArray()) {
+            final int closestOtherState = MathUtils.maxElementIndex(IntStream.range(0, K)
+                    .mapToDouble(n -> n == state ? Double.NEGATIVE_INFINITY : -Math.abs(hiddenStateValues.get(n) - hiddenStateValues.get(state)))
+                    .toArray());
+            final boolean hasLowWeight = weights[state] < MAX_WEIGHT_CONSIDERED_FOR_PRUNING;
+            final boolean isCloseToNeighbor = Math.abs(hiddenStateValues.get(state) - hiddenStateValues.get(closestOtherState)) < DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_SPURIOUS;
+            final boolean isVeryCloseToNeighbor = Math.abs(hiddenStateValues.get(state) - hiddenStateValues.get(closestOtherState)) < DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_DEFINITELY_SPURIOUS;
+
+            final boolean hasLessWeightThanNeighbor = weights[state] < weights[closestOtherState];
+
+            final boolean tinyWeight = weights[state] < AUTOMATICALLY_PRUNED_WEIGHT;
+            final boolean parasite = hasLowWeight && isCloseToNeighbor && hasLessWeightThanNeighbor;
+            final boolean isClone = isVeryCloseToNeighbor && hasLessWeightThanNeighbor;
+            if ( tinyWeight || parasite || isClone) {
+                componentsToPrune.add(state);
+            }
+        }
+
+        weights = IntStream.range(0, K)
+                .filter(n -> !componentsToPrune.contains(n)).mapToDouble(n -> weights[n]).toArray();
+        hiddenStateValues = IntStream.range(0, K)
+                .filter(n -> !componentsToPrune.contains(n)).mapToObj(hiddenStateValues::get).collect(Collectors.toList());
+
+        final StringBuilder message = new StringBuilder("After pruning (state, weight) pairs are: ");
+        for (int n = 0; n < weights.length; n++) {
+            final double weight = weights[n];
+            message.append(String.format("(%f, %f)", hiddenStateValues.get(n), weights[n]) + ((n < weights.length - 1) ? "; " : "."));
+        }
+        logger.info(message);
```

Comment:
did the basic reporting you requested, more information in this method would be awkward.  I think that I will soon overhaul this code to the point that tweaking the logging more is premature.